Ajoute une numérotation visuelle aux titres

artragis opened the pull request:
Fix #6228 

La PR ajoute une numérotation visuelle sur le site web comme sur le PDF : 
![hierarchy](https://user-images.githubusercontent.com/1523653/180873129-43613d7b-cb39-4157-8d17-47d1e6d68563.png)

Pour l'instant le seul cas aux limites ce sont les h3/h4 dans les introductions de tuto/parties et de ce fait dans les billets rédigés dans l'intro

### Contrôle qualité

Builder le front et refresh le cache

Aller dans un big tuto, visiter un de ses chapitres 
Aller dans un mini tuto
Aller dans un article
Aller dans un billet

A chaque fois lorsqu'on met des titres, on a des numérotations.
<!-- Merci ! -->

artragis committed:
Ajoute une numérotation visuelle aux titres

Arnaud-D commented:
QA NOK :heavy_multiplication_x: 

Sur les titres de section et parties, on a un soucis d'espacement.

![image](https://user-images.githubusercontent.com/35631001/184475131-861d2593-ba6c-493f-ada1-de44729d4bc3.png)

Il est aussi présent là, mais c'est plus secondaire :

![image](https://user-images.githubusercontent.com/35631001/184475220-09fcd3e3-2ba2-4e23-a5cf-16d98afd5a1d.png)

La numérotation est collée au texte et c'est pas propre.

L'espace est bien présent dans le code, mais il est bouffé au rendu quelque part... Ça semble lié au fait qu'il y a un lien sur le titre, donc la situation n'est pas la même que les titres dans le corps de texte du point de vue du `:before`.

Arnaud-D reviewed (changes requested):
Voir message précédent.

AmauryCarrade committed:
[numérotation] Factorisation du SCSS ; style.

- Inutile de répéter les sélecteurs, on fait du SCSS, on peut imbriquer.
- Correction du style de code d'après les règles editorconfig : indentation avec deux espaces.

AmauryCarrade committed:
[numérotation] Correction des soucis d'espacement.

- On ne s'appuie plus sur des espaces dans le contenu, mais sur une marge à la
  taille relative à la largeur d'une lettre de la police courante.
- Les compteurs sont réinitialisés en une ligne (plus clair, inutile de
  s’embêter).

AmauryCarrade committed:
[numérotation] Tentative de mise en avant du titre

- Les numérotations sont à peine plus claires.
- J'ai tenté aussi de retirer la graisse. Je ne sais que choisir.

AmauryCarrade commented:
On m'a dit qu'il y avait un souci de CSS.

J'ai regardé et c'était assez facile à régler : au lieu de se baser sur les espaces dans `content`, on met une marge (dont la taille est relative à la largeur des lettres de la police courante), et ça roule.

J'ai, au passage, mis à jour le code depuis la branche principale et refactorisé le CSS & SCSS aux bonnes pratiques (code imbriqué, propriétés non-dupliquées).

## Essais de mise en valeur du texte des titres

J’ai tenté quelque chose pour que le numéro du titre ne soit pas au même niveau d'importance visuelle que le texte. J'ai, dans cette PR, diminué légèrement l'opacité des numéros. Mais j'avais aussi tenté de diminuer la graisse. Voici les deux rendus, à la suite, après le rendu de base. Qu'en pensez-vous ?

![Titres de différents niveaux numérotés (1, 1.1, etc.). Les numéros des titres sont de la même couleur et graisse que le reste.](https://github.com/zestedesavoir/zds-site/assets/1417570/9d39da0c-4101-46e9-9d32-d8a72a41389b)
_Version de base_

![Titres de différents niveaux numérotés (1, 1.1, etc.). Les numéros des titres sont de la même graisse mais un peu plus clairs que le reste.](https://github.com/zestedesavoir/zds-site/assets/1417570/0e138ad7-6ce2-4feb-b5f3-4ad9af214589)
_Version opacité réduite_

![Titres de différents niveaux numérotés (1, 1.1, etc.). Les numéros des titres sont de la même couleur mais avec moins de graisse (plus fins) que le reste.](https://github.com/zestedesavoir/zds-site/assets/1417570/630b99a9-8741-4c83-9348-92acb9e42773)
_Version graisse réduite_

## Souhaitable partout ?

Je pose au passage la question. Veut-on numéroter ainsi _tous_ les titres de _tous_ les contenus ? Ne serais-ce pas un peu lourd pour rien, notamment sur des contenus souvent légers comme les billets, articles, ou mini-tutos ?

Ne serait-ce pas mieux de limiter cette numérotation aux moyens et gros contenus ?

## ⚠️ Note de QA supplémentaire

Ce code CSS ajoutant la numérotation _est assez peu spécifique_ (au sens CSS du terme : il pourrait s'appliquer facilement à des titres hors des articles). J'ajoute donc aux instructions de contrôle qualité le point suivant (et un autre qui n'est pas lié mais qui est aussi important).

- Vérifier ailleurs sur le site qu'aucun autres titre hors des contenus n'est numéroté par accident.
- Sur le sommaire d'un gros tuto (première page), vérifier que la numérotation n'est pas dupliquée.

AmauryCarrade commented:
Ça vient avec quelques soucis de rendu de titres sur les pages principales des sections de gros tutos, où faute de parents, les titres sont numérotés 0 (même si on est dans la partie 2, par exemple).

<img width="562" alt="Sommaire d'une partie où les titres sont numérotés 0.1, 0.1.1, et 0.1.2." src="https://github.com/zestedesavoir/zds-site/assets/1417570/0bb8cb71-b1e5-4531-91c8-8cb8807bc91e">


Pour être honnête, j'ai un peu peur que cette fonctionnalité amène plus de problème subtils et un peu pénibles à résoudre proprement, comme ceci, qu'elle n'apporte de confort de repérage…

AmauryCarrade commented:
Autres soucis repérés

- L'ancienne page _À propos_ est numérotée (ceci _devrait_ être corrigé avec la nouvelle page _À propos_)
  ![Les titres Droits, Système/backend et Front-end de la page À propos sont numérotés en 0.1, 0.2, et 0.3.](https://github.com/zestedesavoir/zds-site/assets/1417570/88be7ade-1c47-43b4-b1a2-f3319868c3dc)
- Les CGU sont doublement numérotées
  ![Les titres de chaque section des CGU sont numérotés en 0.1, 0.2, etc. En plus, les titres sont re-numérotés car ils le sont à la main dans le HTML.](https://github.com/zestedesavoir/zds-site/assets/1417570/1e8609e9-979c-46d9-aac1-ca542c9ec019)
- Les pages sommaires de parties ou chapitres sont doublement numérotées (car ce sont des listes numérotées de base)
  ![Dans le sommaire d'un chapitre (ici), les pages du chapitres sont numérotées en 1.1, 2.1, 2.2 ET en 1, 2, etc. (la numérotation recommençant pour chaque chapitre). Les sections des pages ne sont pas concernées par ce problème et sont correctement numérotées une fois.](https://github.com/zestedesavoir/zds-site/assets/1417570/4b8bbf56-d1f6-4f3f-a369-a646646575af)
 

Je soupçonne que toutes les pages de `/pages` sont concernées de près ou de loin.

Arnaud-D commented:
Pour moi, c'est bloquant la numérotation à partir de zéro dans l'introduction des tutos (et dans la conclusion aussi ?). Il y a énormément de contenus avec de genre de choses et ça fait vraiment buggué d'avoir ce genre de numérotation.

AmauryCarrade commented:
Honnêtement, je trouve que tous les exemples de défauts que j'ai donné sont bloquants, pas que ceux des sommaires et conclusions (oui, en effet) de parties.

Ça risque d'être particulièrement _tricky_, avec le code actuel.

Arnaud-D commented:
@philippemilink J'ai l'impression que le consensus est que c'est piégeux. On ferme ?

artragis commented:
Allez, zou
